### PR TITLE
Rework newTab() utility to work with 'noopener'.

### PR DESCRIPTION
**Pull request by whiten**
This reimplements its internal message system to use localStorage events
rather than postMessage(). This does not mean that postMessage can't be
used for tests that pass {opener: true} to newTab, but makes the
underlying mechanism work regardless.

### CODE REVIEW

File: src/test-util.ts [hunk 1]
```diff
@@ -72,21 +71,23 @@ type LogResponse = {
...
-  const {opener} = window;
-  if (opener === null) {
-    return;
-  }
+export async function initializeNewTab(
+  tabId: string,
+  path: string,
+  event: StorageEvent,
+): Promise<void> {
+  const send = (message: unknown) => {
```
Comment (arv) [inline src/test-util.ts]: We have been using `JSONValue` for these but unknown is OK too.

File: src/testutil-new-tab.html [hunk 2]
```diff
@@ -1,12 +1,17 @@
 <html>
   <head>
     <script type="application/javascript">
```
Comment (arv) [inline src/testutil-new-tab.html]: Remove type attribute. JS is default

File: src/testutil-new-tab.html [hunk 3]
```diff
@@ -1,12 +1,17 @@
 <html>
```
Comment (arv) [inline src/testutil-new-tab.html]: I missed this file last time :-)

Add `<!doctype html>`

File: src/testutil-new-tab.html (hunk 2 above)
Comment (arv) [reply src/testutil-new-tab.html]: maybe even use `type="module"`which opts you into strict module mode.

File: src/web-lock.test.ts [hunk 4]
```diff
@@ -4,8 +4,13 @@ import {newTab, Tab} from './test-util';
 
 let tab: Tab;
 
+function isFirefox(): boolean {
+  return navigator.userAgent.toLowerCase().indexOf('firefox') > -1;
```
Comment (arv) [inline src/web-lock.test.ts]: Use a regxp?

```suggestion
  return /firefox/i.test(navigator.userAgent);
```

File: src/test-util.ts (hunk 1 above)
Comment (arv) [inline src/test-util.ts]: We have been using `JSONValue` for these but `unknown` is OK too.

**Review by arv — APPROVED** [review]

File: src/test-util.ts (hunk 1 above)
Comment (whiten) [reply src/test-util.ts]: Switched this to `JSONObject`, and used `JSONValue` in a few other places. Thanks!

File: src/testutil-new-tab.html (hunk 2 above)
Comment (whiten) [reply src/testutil-new-tab.html]: Thanks, done.

File: src/testutil-new-tab.html (hunk 3 above)
Comment (whiten) [reply src/testutil-new-tab.html]: Done.

File: src/web-lock.test.ts (hunk 4 above)
Comment (whiten) [reply src/web-lock.test.ts]: Good call. I definitely didn't use StackOverflow for this. ;)

File: src/test-util.ts (hunk 1 above)
Comment (whiten) [reply src/test-util.ts]: Done.

File: src/testutil-new-tab.html (hunk 3 above)
Comment (whiten) [reply src/testutil-new-tab.html]: I also renamed this file to `src/test-util-new-tab.html` to match the punctuation on `src/test-util.ts`.

**Pull request merged**